File: app/src/main/java/com/crushcoder/movie/service/ApiService.java
package com.crushcoder.movie.service;


import com.crushcoder.movie.service.response.Movie;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {
    @GET("3/movie/{categories}")
    Single<List<Movie>> getMoviesInfo(@Path("categories") String categories,
                                      @Query("page") int page,
                                      @Query("api_key") String apiKey);
}

File: app/src/main/java/com/crushcoder/movie/service/ApiService.kt
package com.crushcoder.movie.service

import com.crushcoder.movie.ui.movie.Movie
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query


interface ApiService {
    @GET("3/movie/{categories}")
    fun getMoviesInfo(@Path("categories") categories: String,
                      @Query("page") page: Int,
                      @Query("api_key") apiKey: String): Call<List<Movie>>

    companion object Factory {
        fun create(): ApiService {
            val retrofit = Retrofit.Builder()
                    .addConverterFactory(GsonConverterFactory.create())
                    .baseUrl("http://api.themoviedb.org/")
                    .build()
            return retrofit.create(ApiService::class.java)
        }
    }
}

